### PR TITLE
created new user registration page and navbar

### PR DESCRIPTION
**Pull request by RobLTMacP**
Changes made:

- Navbar created and added to Layout
- "UserBadge" nested inside NavBar component 
- Routing for Registration page implemented
- UI for Registration and groundwork for backend routing implemented 

Needs still:
- Some UI updates 
- Testing and debugging once DB is up and running

### CODE REVIEW

File: src/components/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <ul className="menu menu-horizontal px-1">
+        <li><a>Profile</a></li>
+        <li>
+          <details>
+            <summary>
+              Main Menu
+            </summary>
+            <ul className="p-2 bg-base-100 rounded-t-none">
+              <li><a>Sign In</a></li>
+              <li><a href='/Register'>Create New Accountt</a></li>
```
Comment (stefan-apr) [inline src/components/NavBar.tsx]: 'Accountt' -> 'Account'

**Review by stefan-apr — CHANGES_REQUESTED** [review]
One small thing, just caught a spelling mistake in there, otherwise looks good. 
If you use VSCode, I definitely recommend the Code Spell Checker extension - it's come in real handy for me in the time that I've used it.

File: src/components/NavBar.tsx (hunk 1 above)
Comment (RobLTMacP) [reply src/components/NavBar.tsx]: Thank you! Added that extension and fixed the error.

**Review by stefan-apr — APPROVED** [review]
Looks good! Approved

**Pull request merged**